app/jobs/email_petition_signatories.rb

```diff
@@ -0,0 +1,85 @@
+class EmailPetitionSignatories
+  class PleaseRetry < StandardError; end
+
+  def self.run_later_tonight(job_class, petition, requested_at, *extra_args)
```

Comment:
Maybe it's just me, but it too quite a long time to understand what's going on here between the jobs, `EmailPetitionSignatories` and the `Worker`.

Somehow I feel that `self.run_later_tonight` and `self.later_tonight` should not be the responsibility of this class.

What do you think if this class would act just as the worker - and those two methods could be extracted to a separate class, something like `EmailPetitionSignatoriesJob`, which would be a superclass for `EmailDebateOutcomesJob` and `EmailDebateOutcomesJob`? The jobs then would no longer have to communicate with this class to queue themselves.